openwebwork/pg
# Splitting webwork2 code from PG

This PR is designed to remove all webwork2 code from the PG side.   You need openwebwork/webwork2#1463 as well. 

There are two main parts to this

1. WWSafe.pm (from webwork2) is moved to PG_ROOT/lib. 
2. A PGEnvironment module is created to mimic some of the functionality of CourseEnvironment.  

This will also better testing of PG code without needing any webwork2 modules. 

To Test if works with current webwork2:

1. check out this PR and openwebwork/webwork2#1463
2. The top of `WEBWORK_ROOT/conf/webwork.apache2.4.config` needs to be updated.  `pg_dir` is needed to be defined here and in `use lib ...`  See the .dist file for the changes. 
3. restart apache.

Test some problems including those with  tikz and other plotting that create images.  These use the `IO.pm` file, which has been rewritten.  

I'll post some other documentation here about testing some pg code that demonstrates the split from webwork2.
